starlette/routing.py

```diff
@@ -108,34 +108,47 @@ def replace_params(
 
 
 def compile_path(
-    path: str,
+    pattern: str,
```

Comment:
Question, isn't this still considered backwards-incompatible?

Reply:
Ugh. I guess technically you are correct. I didn't think about keyword argument usage.
We are also on a 0.x version. So we could do breaking changes in 0.21 I guess.
How about defining an alias for now. Then we can also get rid of the inprecise `compile_path` name internally.

E.g. rename into `def compile_pattern(pattern: str)` and `def compile_path(path: str): return compile_pattern(path)` for backwards compatibility.

Reply:
Indeed adding a parameter is a breaking change

Reply:
Imho the method should not have been used for compiling host patterns in the first place, because a hostname is not a path.
The least we can do now is apply proper naming and documentation.

The renaming would allow keeping backward-compatibility for the old interface, while implementing a better on. Personally I wouldn’t regard this function as part of the public starlette api currently, because it is not documented anywhere and I would keep it that way. So the new function name would be `_compile_pattern` I guess.

Having said all that, I don’t feel too strongly about any of this. Just thought it would be an improvement to the code base. If you don’t care about correct naming, I can simply revert that part.

Reply:
yes I think this has come up a few times before, but the priority is to keep backwards-compatibility until there's a major release like from 1.x.x to 2.x.x

Reply:
The only case when this would break is if people do `compile_path(path=...)`.

It's possible to go for a deprecation, allowing this usage at runtime while keeping the new signature for type hints, with a decorator...

```python
def _allow_deprecated_path_kwarg(func):
    @wraps(func)
    def wrapped(*args: Any, **kwargs: Any):
        if "path" in kwargs:
            ...  # Raise a deprecation warning
            return func(path)
        return func(*args, **kwargs)

    return wrapped

@_allow_deprecated_path_kwarg
def compile_path(pattern: str):
    ...
```